File: CarHomeWidget/src/main/java/com/anod/car/home/prefs/views/SeekBarDialogPreference.java
package com.anod.car.home.prefs.views;

import android.content.Context;
import android.content.res.TypedArray;
import android.support.v7.preference.DialogPreference;
import android.support.v7.preference.PreferenceManager;
import android.util.AttributeSet;

import com.anod.car.home.R;

/**
 * @author alex
 * @date 2015-09-14
 */
public class SeekBarDialogPreference extends DialogPreference {
    private static final String ANDROIDNS = "http://schemas.android.com/apk/res/android";


    private final String mSuffix;

    private final int mMax;

    private int mValue;

    public SeekBarDialogPreference(Context context, AttributeSet attrs) {
        this(context, attrs, R.attr.dialogPreferenceStyle);
    }

    public SeekBarDialogPreference(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);

        mSuffix = attrs.getAttributeValue(ANDROIDNS, "text");
        mMax = attrs.getAttributeIntValue(ANDROIDNS, "max", 100);
    }


    public int getMax() {
        return mMax;
    }

    public void setValue(int value) {
        this.mValue = value;
    }

    public int getValue() {
        return mValue;
    }

    public String getSuffix() {
        return mSuffix;
    }

    public boolean persistInt(int value) {
        return super.persistInt(value);
    }

    @Override
    protected Object onGetDefaultValue(TypedArray a, int index) {
        return a.getInt(index, 0);
    }

    @Override
    protected void onSetInitialValue(boolean restore, Object defaultValue) {
        mValue = getPersistedInt(defaultValue == null ? 0 : (Integer) defaultValue);
    }

    protected void onAttachedToHierarchy(PreferenceManager preferenceManager) {
        super.onAttachedToHierarchy(preferenceManager);
    }
}

File: CarHomeWidget/src/main/java/com/anod/car/home/prefs/views/SeekBarDialogPreference.kt
package com.anod.car.home.prefs.views

import android.content.Context
import android.content.res.TypedArray
import android.util.AttributeSet
import androidx.preference.DialogPreference
import com.anod.car.home.R

/**
 * @author alex
 * @date 2015-09-14
 */
class SeekBarDialogPreference @JvmOverloads constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int = R.attr.dialogPreferenceStyle) : DialogPreference(context, attrs, defStyleAttr) {

    val suffix = attrs.getAttributeValue(ANDROIDNS, "text") ?: ""
    val max = attrs.getAttributeIntValue(ANDROIDNS, "max", 100)
    var value: Int = 0

    public override fun persistInt(value: Int): Boolean {
        return super.persistInt(value)
    }

    override fun onGetDefaultValue(a: TypedArray?, index: Int): Any {
        return a!!.getInt(index, 0)
    }

    override fun onSetInitialValue(defaultValue: Any?) {
        value = getPersistedInt(defaultValue as? Int ?: 0)
    }

    companion object {
        private const val ANDROIDNS = "http://schemas.android.com/apk/res/android"
    }
}
